Public/Src/Engine/Scheduler/Artifacts/FileContentManager.cs

```diff
@@ -1341,23 +1342,25 @@ private async Task<Possible<Unit>> TryHashFileArtifactsAsync(PipArtifactsState s
                     {
                         // Directory artifact contents are not hashed since they will be hashed dynamically
                         // if the pip accesses them, so the file is the declared artifact
-                        state.HashTasks.Add(TryQueryContentAsync(
+                        var artifactContentInfo = await TryQueryContentAsync(
                             file,

```

Comment:
Is it intentional to remove the parallelism here?

Reply:
Fixed in new commit